Src/AutoFixture/MailAddressGenerator.cs

```diff
@@ -36,10 +36,21 @@ public object Create(object request, ISpecimenContext context)
                 return new NoSpecimen(request);
             }
 
-            var name = Guid.NewGuid().ToString();
-            var host = fictitiousDomains[(uint)name.GetHashCode() % 3];
+            if (context == null)
+            {
+                return new NoSpecimen(request);
+            }
+                    
+            var localPart = context.Resolve(typeof(EmailAddressLocalPart)) as EmailAddressLocalPart;
+
+            if (localPart == null)
+            {
+                return new NoSpecimen(request);

```

Comment:
(This implementation is correct - don't change it.)

Here it becomes really strange, but as far as AutoFixture is concerned, `null` is actually a valid return value... :sweat: That's not by any of my standard design principles, but because AutoFixture is attempting to be useful for more people than only me, and sometimes people want to be able to configure AutoFixture to return null in certain cases.

(This is the underlying reason that the `NoSpecimen` signal type exists in the first place. A Builder can't use `null` to signal that it doesn't want to handle the request, because it would then be impossible to distinguish between a `null` returned as an opt-out, and `null` returned as a *value*.)

If the returned `localPart` is null, it's not an error of the *caller* of the Builder, so at this point the preconditions are still satisfied: it would be wrong to throw an exception. Instead, it's correct to return `NoSpecimen` so that a potential fallback builder can get a chance to handle the request.